src/resource/util.cpp

```diff
@@ -165,13 +170,44 @@ void iterate_files_worker(const string& path,
         closedir(dir);
     }
 }
+#endif
 
 void iterate_files(const string& path,
                    std::function<void(const string& file, bool is_dir)> func,
                    bool recurse)
 {
     vector<string> files;
     vector<string> dirs;
+#ifdef _WIN32
+    string file_match = path_join(path, "*");
+    WIN32_FIND_DATA data;
+    HANDLE hFind = FindFirstFile(file_match.c_str(), &data);
+    if (hFind != INVALID_HANDLE_VALUE)
+    {
+        do
+        {
+            bool is_dir = data.dwFileAttributes == FILE_ATTRIBUTE_DIRECTORY;
+            if (is_dir)
+            {
+                if (string(data.cFileName) != "." && string(data.cFileName) != "..")
+                {
+                    string dir_path = path_join(path, data.cFileName);
+                    if (recurse)
+                    {
+                        iterate_files(dir_path, func, recurse);
+                    }
+                    func(dir_path, true);
+                }
+            }
+            else
+            {
+                string file_name = path_join(path, data.cFileName);
+                func(file_name, false);
+            }
+        } while (FindNextFile(hFind, &data));
+        FindClose(hFind);
+    }
+#else
```

Comment:
Use a separate if for the non-windows platforms.

Reply:
I don't get it. The "#else" part is handling non-windows platforms.
I don't know a way to express "if (APPLE or LINUX)" like we do in cmake for a c/c++ preprocessor.